### PR TITLE
DDCE-2642 : Add scalafmt + update libs + code refactor

### PR DESCRIPTION
**Pull request by lmugnai**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import play.core.PlayVersion
+import sbt._
+
+object AppDependencies {
+  private lazy val silencerVersion                   = "1.7.9"
+  private lazy val monocleVersion       = "1.7.3"
+  private lazy val bootstrapPlayVersion = "7.0.0"
+  private lazy val hmrcMongoPlayVersion = "0.68.0"
```
Comment (frikit) [inline project/AppDependencies.scala:8]: if we are doing full verison bump please update this to 0.71.0 as well

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import play.core.PlayVersion
+import sbt._
+
+object AppDependencies {
+  private lazy val silencerVersion                   = "1.7.9"
```
Comment (frikit) [inline project/AppDependencies.scala:5]: I can see this file is not formatted with scalafmt please do

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "uk.gov.hmrc.mongo"            %% "hmrc-mongo-play-28"        % hmrcMongoPlayVersion,
+    "org.scalaj"                   %% "scalaj-http"               % "2.4.2",
+    "com.fasterxml.jackson.module" %% "jackson-module-scala"      % "2.13.3",
+    compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
+    "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
+  )
+
+  private val test: Seq[ModuleID] = Seq(
+    "uk.gov.hmrc.mongo"      %% "hmrc-mongo-test-play-28" % hmrcMongoPlayVersion,
+    "org.scalatest"          %% "scalatest"               % "3.2.12",
```
Comment (frikit) [inline project/AppDependencies.scala:23]: can be 3.2.13

**Review by frikit — CHANGES_REQUESTED** [review]

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,16 +1,12 @@
 resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefacts.tax.service.gov.uk/maven2")
-resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
-
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.6.0")
+resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(
+  Resolver.ivyStylePatterns
+)
 
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.8.0")
```
Comment (frikit) [inline project/plugins.sbt:6]: I can see this file is not formatted as well

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import play.core.PlayVersion
+import sbt._
+
+object AppDependencies {
+  private lazy val silencerVersion                   = "1.7.9"
+  private lazy val monocleVersion       = "1.7.3"
+  private lazy val bootstrapPlayVersion = "7.0.0"
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply project/AppDependencies.scala]: done

File: project/AppDependencies.scala (hunk 1 above)
Comment (lmugnai) [reply project/AppDependencies.scala]: done

File: project/AppDependencies.scala (hunk 2 above)
Comment (lmugnai) [reply project/AppDependencies.scala]: done

File: test/it/helpers/BaseSpec.scala [hunk 6]
```diff
@@ -27,17 +27,25 @@ import org.scalatest.featurespec.AnyFeatureSpec
...
 
+  implicit override lazy val app: Application = GuiceApplicationBuilder()
+    .configure(
+      "auditing.enabled"       -> false,
+      "auditing.traceRequests" -> false,
+      "mongodb.uri"            -> mongoUri,
+      "run.mode"               -> "It"
+    )
+    .build()
+  // scalastyle:off magic.number
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply test/it/helpers/BaseSpec.scala]: done

File: project/plugins.sbt (hunk 4 above)
Comment (lmugnai) [reply project/plugins.sbt]: done

**Review by frikit — APPROVED** [review]

**Pull request merged**